### DISCUSSION TITLE
Paperless-ngx update issue

### DISCUSSION THREAD

**Discussion by DimonXe** [tteck/Proxmox#2537] [Show and tell]
Hi. Couldn't find anywhere else to post, so I created a separate. I apologise if I did it wrong.

The problem is the following: after a few updates, my scheduler stops working and paperless no longer picks up mail and trains its AI.
It doesn't happen after every update, but periodically.
In the logs everything is fine, there are no errors. If you upload files manually or from a shared folder, everything works fine too.

If you run “systemctl status paperless-scheduler.service” you can see that the scheduler is not working

```
root@paperless-ngx:/var/log# sudo systemctl status paperless-scheduler.service
* paperless-scheduler.service - Paperless Celery beat
     Loaded: loaded (/etc/systemd/system/paperless-scheduler.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Thu 2024-02-22 08:49:23 CET; 26s ago
    Process: 23281 ExecStart=celery --app paperless beat --loglevel INFO (code=exited, status=1/FAILURE)
   Main PID: 23281 (code=exited, status=1/FAILURE)
        CPU: 1.544s
```

After a long search I found a solution:
To get mailer working normally again you need to rename or delete the old database and restart the scheduler service:

```
cd /opt/paperless/data
mv celerybeat-schedule.db.db celerybeat-schedule.db.db.bak
sudo systemctl start paperless-scheduler.service
```

After that, the scheduler starts working normally again.

To restore document classifier, you need to do the following:

```
cd /opt/paperless/src
python3 manage.py document_create_classifier
```

It didn't work for me once and I had to re-download the dictionaries again, but that was only the case when the problem first appeared:

```
mkdir /usr/share/nltk_data
python3 -m nltk.downloader stopwords
python3 -m nltk.downloader punkt
```

I haven't found a description of this problem and I don't understand why only I've encountered it, but maybe it will help someone else too.

**Comment by tteck** [comment]
Thank you for the write-up!

**Comment by rgb2017** [comment]
Same issue here. I've installed Paperless-ngx using your script and the Celery service is unhealthy after a fresh install and after restarts. I did confirm with three different installs the same behavior exists. I did also see this with the classifier. Running update and selecting to apply an update fixes it temporarily until the next restart. Any way to get an update that fixes this so it starts up correctly after reboot?
